dask/array/reductions.py

```diff
@@ -349,26 +385,37 @@ def nannumel(x, **kwargs):
     return chunk.sum(~np.isnan(x), **kwargs)
 
 
-def mean_chunk(x, sum=chunk.sum, numel=numel, dtype='f8', **kwargs):
+def mean_chunk(x, sum=chunk.sum, numel=numel, dtype='f8', meta=False, **kwargs):
     n = numel(x, dtype=dtype, **kwargs)
+    if meta:
```

Comment:
Is meta a boolean here or possibly a numpy array?  If the latter, then beware that bool calls on numpy arrays aren't always supported

```python
In [1]: import numpy as np

In [2]: x = np.ones((5, 5))

In [3]: if x: print("hello")
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-74881c0c1cf7> in <module>
----> 1 if x: print("hello")

ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

Reply:
In this case, meta is always a boolean. Its use is merely to identify that the input array is a meta array, this is necessary in https://github.com/dask/dask/pull/4543/files#diff-0163583b023e0f8cb67c20c17374d557R242 to avoid complex computations on those arrays, and in special to avoid returning dicts that we don't want to handle, since that doesn't provide any necessary information for meta.